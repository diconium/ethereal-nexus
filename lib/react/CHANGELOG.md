# @ethereal-nexus/react

## 0.2.0

### Minor Changes

- 90cf802: Update dependencies

## 0.1.2

### Patch Changes

- 205469c: Changed slot to allow the prop drill to nexus components inside experience fragments

## 0.1.1

### Patch Changes

- f882265: Check for cq-Editable-dom--container instead of section to spread the properties into a slot to fix issues when the page is not reloaded

## 0.1.0

### Minor Changes

- a367679: feat: add Slot component that enables drilling props to slotted nexus components
