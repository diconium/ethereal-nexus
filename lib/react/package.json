{
  "name": "@ethereal-nexus/react",
  "version": "0.1.1",
  "description": "",
  "keywords": [
    "react"
  ],
  "author": "",
  "license": "Apache-2.0",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "build": "tsup ./src",
    "dev": "tsup ./src --watch"
  },
  "types": "./dist/index.d.ts",
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "require": "./dist/index.js",
      "import": "./dist/index.mjs"
    }
  },
  "sideEffects": false,
  "files": [
    "dist",
    "README.md"
  ],
  "devDependencies": {
    "@types/react": "^19.0.8",
    "tsup": "^8.0.2",
    "typescript": "^5.6.2"
  },
  "peerDependencies": {
    "react": "^18.0.0 || ^19.0.0",
    "react-dom": "^18.0.0 || ^19.0.0"
  }
}
