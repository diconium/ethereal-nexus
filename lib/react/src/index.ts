export { Slot } from './Slot/Slot';