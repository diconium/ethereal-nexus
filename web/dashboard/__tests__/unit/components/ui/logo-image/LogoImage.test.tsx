import React from 'react';
import { RenderResult, render, screen } from '@testing-library/react';
import LogoImage from '@/components/ui/logo-image';

describe('LogoImage component', () => {
  let wrapper: RenderResult;

  const renderComponent = () => {
    wrapper = render(<LogoImage/>);
  };

  beforeEach(() => {
    renderComponent();
  });

  it('should match snapshot', () => {
    expect(wrapper.baseElement).toMatchSnapshot();
  });

  it('should render image logo with src and alt attributes', () => {
    const image = screen.getByTestId('ethereal-logo');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', expect.stringContaining('dark'));
    expect(image).toHaveAttribute('alt', 'Ethereal Nexus Logo');
  });
});
