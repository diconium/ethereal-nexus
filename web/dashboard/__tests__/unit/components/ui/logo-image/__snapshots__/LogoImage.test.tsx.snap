// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoImage component should match snapshot 1`] = `
<body>
  <div>
    <img
      alt="Ethereal Nexus Logo"
      class="filter invert hue-rotate-180 dark:filter-none"
      data-nimg="1"
      data-testid="ethereal-logo"
      decoding="async"
      fetchpriority="high"
      height="0"
      sizes="100vw"
      src="/_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=640&q=75 640w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=750&q=75 750w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=828&q=75 828w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fethereal-nexus-logo-dark-mode.jpg&w=3840&q=75 3840w"
      style="color: transparent;"
      width="267"
    />
  </div>
</body>
`;
